spec/faraday/tracer_spec.rb

```diff
@@ -16,11 +16,27 @@
     expect(span.tags['span.kind']).to eq('client')
   end
 
+  describe 'parent_span' do
+    it 'allows to pass a pre-created parent span' do
+      parent_span = tracer.start_span("parent_span")
+      expect(tracer).to receive(:start_span).with(any_args, hash_including(child_of: parent_span)).and_call_original
+      call(method: :post, span: parent_span)
+    end
+
+    it 'allows to pass a block as a parent span provider' do
+      parent_span = tracer.start_span("parent_span")
+      parent_span_provider = lambda { parent_span }
+
+      call(method: :post, span: parent_span_provider)
```

Comment:
Do you think we should also check that the new span has the correct parent id? I think you can use `tracer.finished_spans.last.parent_id` when you make the mock Tracer to accept it ( https://github.com/iaintshine/ruby-faraday-tracer/blob/d9dfbf033b512b87fe8a7709f9c8278dce38f045/spec/faraday/support/recording_tracer.rb#L8 )